[arnav12344] iP

arnav12344 opened the pull request:
### Kayo
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://dansilvestre.com/productivity-quotes/)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

arnav12344 committed:
Added skeleton code

arnav12344 committed:
Added echo feature

arnav12344 committed:
Task class added and implemented

arnav12344 committed:
task 2 implementation

arnav12344 committed:
deadline todo and events added

arnav12344 committed:
added semi-automated testing

arnav12344 committed:
level 5 error handling basic

arnav12344 committed:
Added delete command

arnav12344 committed:
Add storage and retrival

arnav12344 committed:
Store deadline date as dateTime

arnav12344 committed:
Merge branch 'branch-Level-8'

arnav12344 committed:
Add oop to scripts

arnav12344 committed:
Merge branch 'A-More-Oop'

arnav12344 committed:
Make a package

arnav12344 committed:
Merge branch 'Packages'

arnav12344 committed:
Merge branch 'add-gradle-support'

arnav12344 committed:
Add gradle support

arnav12344 committed:
Add Junit tests

arnav12344 committed:
Add jar files

arnav12344 committed:
Add JavaDoc comments

arnav12344 committed:
Remove coding standard violations

arnav12344 committed:
Add find command

arnav12344 committed:
Merge branch 'branch-A-JavaDoc'

arnav12344 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Kayo/Kayo.java

arnav12344 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Kayo/Kayo.java

arnav12344 committed:
Resolve merge conflicts

arnav12344 committed:
Add file

arnav12344 committed:
Merge file

arnav12344 committed:
Revert "Merge branch 'branch-A-CodingStandard'"

This reverts commit 24127906dce20e6782702ef58241d42eb95ac3bf, reversing
changes made to 6652c2cd569c43888d8a77c0cf9f1c99040ec3ce.

arnav12344 committed:
Resolve merge conflicts

arnav12344 committed:
Fix merge errors

pradyuprasad commented on src/main/java/Kayo/Kayo.java:
```diff
@@ -0,0 +1,317 @@
+package Kayo;
```
I noticed the package name uses an uppercase "K." Perhaps kayo in all lowercase would better align with Java naming conventions?

arnav12344 replied:
Thank you for your feedback!

verakohh commented on src/main/java/Kayo/Kayo.java:
```diff
@@ -0,0 +1,317 @@
+                ui.markTask(listTasks.get(index));
+            }else if (splitList[0].equals("todo")) {
+                if (splitList.length==1) {
+                    new DukeException("OOPS !! The description of a todo can't be empty");
```
Is there a reason why its called DukeException if your chatbot's name is Kayo?

verakohh reviewed (commented):
Overall good naming, just one nit about the name of the exception class

pradyuprasad commented on src/main/java/Kayo/Kayo.java:
```diff
@@ -0,0 +1,317 @@
+         * @return List of tasks from database
+         * */
+        public List<Task> load () {
+            List<Task> listTasks = new ArrayList<>();
```
listTasks sounds like a verb. Perhaps the variable name in this (and other places) could be taskList instead

pradyuprasad commented on src/main/java/Kayo/Kayo.java:
```diff
@@ -0,0 +1,317 @@
+    /**
+     * Handles Exceptions in Chatbot
+     */
+    public static class DukeException {
```
I agree with the other commentor that this class should not be named DukeException as your project is called Kayo

pradyuprasad commented on src/main/java/Kayo/Kayo.java:
```diff
@@ -0,0 +1,317 @@
+            }
+            return listTasks;
+        }
+        private Task getTask(String readString) {
```
good job on naming all methods correctly!

arnav12344 replied:
Thank you!

pradyuprasad reviewed (commented):
Overall looks good to me. Minor suggestion is to split the classes into separate files for readability

arnav12344 committed:
Add Gui to chatbot

arnav12344 committed:
Merge branch 'master' of https://github.com/arnav12344/ip

* 'master' of https://github.com/arnav12344/ip:
  Resolve merge conflicts
  Revert "Merge branch 'branch-A-CodingStandard'"
  Merge file
  Add file
  Resolve merge conflicts

# Conflicts:
#	data/kayo.txt
#	src/main/java/Kayo/Kayo.java

arnav12344 committed:
Add assertions

Assert statements help in preventing programmer bug

Lets add assertions to improve the future code quality.

Using assertions over exceptions in this case is preferable as assertions can be disabled during release time

arnav12344 committed:
Kayo run class method: extract multiple methods

For better readability, the run method is too long and does not follow SLAP framework

Let's extract methods to find from list, make tasks etc

Changing for command inheritance will take too much time

arnav12344 committed:
Merge pull request #1 from arnav12344/branch-A-Assertions

Add assertions

arnav12344 committed:
Merge branch 'master' into branch-A-CodeQuality

arnav12344 committed:
Merge pull request #2 from arnav12344/branch-A-CodeQuality

Branch a code quality

arnav12344 committed:
Add help page

arnav12344 committed:
Add forgotten tags

arnav12344 committed:
Update README.md

arnav12344 committed:
Update README.md

arnav12344 committed:
Add Ui.png

arnav12344 committed:
Add Ui.png
